cl/disclosures/tasks.py

```diff
@@ -425,119 +370,72 @@ def import_disclosure(self, data: Dict[str, Union[str, int, list]]) -> None:
-
     interface = make_redis_interface("CACHE")
     disclosure_key = make_disclosure_key(data["id"])
     newly_enqueued = create_redis_semaphore(
         interface,
         disclosure_key,
-        ttl=60 * 60 * 12,
+        ttl=60 * 60 * 2,
     )
 
     if not newly_enqueued:
-        logger.error(
+        logger.info(
             f"Process is already running {data['id']}. {disclosure_key}",
         )
         return
 
-    # Generate PDF content from our three paths
-    year = int(data["year"])
-    person_id = data["person_id"]
-    report_type = data.get("report_type", -1)
-
     logger.info(
-        f"Processing row {data['id']} for person {person_id} "
-        f"in year {year}"
+        f"Processing row {data['id']} for person {data['person_id']} "
+        f"in year {data['year']}"
     )
+    response = requests.get(data["url"], timeout=60 * 20)
 
-    # Check if we've already extracted
-    disclosure_url = get_aws_url(data)
-    was_previously_pdfed = has_been_pdfed(disclosure_url)
-    pdf_response = generate_or_download_disclosure_as_pdf(
-        data, was_previously_pdfed
-    )
-    pdf_bytes = pdf_response.content
-
-    if pdf_response.status_code != 200:
-        logger.error(msg="PDF generation failed.")
+    if response.status_code != 200:
+        logger.error(
+            f"Failed to download {data['id']} {data['url']}",
+            extra={"disclosure_id": data["id"]},
+        )
         interface.delete(disclosure_key)
         return
 
-    if was_previously_pdfed:
-        disclosure = get_disclosure_from_pdf_path(disclosure_url)
+    # Check if disclosure already exists
+    query = FinancialDisclosure.objects.filter(download_filepath=data["url"])
+    if len(query) > 0:
+        # If previously uploaded, use disclosure else process new document
+        disclosure = query[0]
     else:
-        logger.info("PDF generated successfully.")
-
-        # Sha1 hash - Check for duplicates
-        sha1_hash = sha1(pdf_bytes)
-        in_system = check_if_in_system(sha1_hash)
-        if in_system:
-            # If we are given duplicate images as different files.  Sadly something to test for.
-            logger.error(
-                "PDF already in system.",
-                extra={"disclosure_id": disclosure_key},
-            )
-            interface.delete(disclosure_key)
-            return
-
-        # Return page count - 0 indicates a failure of some kind.  Like PDF
-        # Not actually present on aws.
-        pg_count = get_page_count(pdf_bytes)
-        if not pg_count:
+        disclosure = save_and_upload_disclosure(
+            interface, disclosure_key, response, data
+        )
+        if not disclosure:
             logger.error(
-                msg=f"Page count failed",
-                extra={"disclosure_id": disclosure_key, "url": disclosure_url},
+                f"Upload to save disclosure {data['id']} {data['url']}",
+                extra={"disclosure_id": data["id"]},
```

Comment:
I think you forgot a verb here or a noun? Not sure. :)

Reply:
√